libbeat/esleg/eslegclient/enc.go

```diff
@@ -19,11 +19,13 @@ package eslegclient
 
 import (
 	"bytes"
-	"compress/gzip"
+
 	"io"
 	"net/http"
 	"time"
 
+	"github.com/klauspost/compress/gzip"
```

Comment:
Is it supposed to be here?

Reply:
so from this discussion - we wanted to go ahead with using this library so https://github.com/elastic/beats/issues/38599#issuecomment-2470975079

Reply:
Just double-checking because the description didn't say we switched to the new library in this PR, I thought it was only about benchmarks.

We might want to have a change log entry for this as well because it has user impact.